File: src/main/java/com/amazonaws/iot/fleetwise/vehiclesimulator/cli/VehicleSimulatorCommand.java
package com.amazonaws.iot.fleetwise.vehiclesimulator.cli;

import java.util.Arrays;
import java.util.concurrent.Callable;

import org.jetbrains.annotations.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;
import picocli.CommandLine;
import picocli.CommandLine.Command;

@Component
@Command(
        subcommands = {UploadToS3Command.class, LaunchVehicles.class, StopVehicles.class, FixFirmwareVersion.class},
        versionProvider = VersionProvider.class,
        mixinStandardHelpOptions = true,
        description = {"Utility for launching simulated vehicles"}
)
public final class VehicleSimulatorCommand implements Callable<Integer> {

    private final Logger log = LoggerFactory.getLogger(VehicleSimulatorCommand.class);


    public Integer call() {
        this.log.info("FleetWise Vehicle Simulator \n");
        this.log.info("To continue, please select one of the following options when running this command: LaunchVehicles, StopVehicles, or FixFirmwareVersion \n");
        return 0;
    }

    public static void main(@NotNull String[] args) {
        System.exit((new CommandLine(new VehicleSimulatorCommand())).execute((String[]) Arrays.copyOf(args, args.length)));
        throw new RuntimeException("System.exit returned normally, while it was supposed to halt JVM.");
    }

}

File: src/main/kotlin/com/amazonaws/iot/vehiclesimulator/cli/VehicleSimulatorCommand.kt
package com.amazonaws.iot.autobahn.vehiclesimulator.cli

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import picocli.CommandLine
import picocli.CommandLine.Command
import java.util.concurrent.Callable
import kotlin.system.exitProcess


class KotlinApp {

    public fun getGreeting() : String {
        return "Hello World from Kotlin!";
    }}

/*fun main() {
    print(KotlinApp().getGreeting())
} */

@Command(
    mixinStandardHelpOptions = true,
    versionProvider = VersionProvider::class,
    description = ["Utility for launching simulated vehicles"],
    subcommands = [
        UploadToS3Command::class,
        LaunchVehicles::class,
        StopVehicles::class,
        FixFirmwareVersion::class
    ]
)

class VehicleSimulatorCommand : Callable<Int> {
    private val log: Logger = LoggerFactory.getLogger(VehicleSimulatorCommand::class.java)
    override fun call(): Int {
        log.info("FleetWise Vehicle Simulator \n")
        log.info("To continue, please select one of the following options when running this command: LaunchVehicles, StopVehicles, or FixFirmwareVersion \n")
        
        return 1
    }

    //companion object {
        //@JvmStatic
      
    //}
}

fun main(args: Array<String>) {
    exitProcess(CommandLine(VehicleSimulatorCommand()).execute(*args))
} 




